docs/experiments/generic-pattern-matching.md

```diff
@@ -137,6 +137,61 @@ name = ...
   name = ...
 ```
 
+### How to handle line-based input
+
+Many simple configuration formats are a collection of key/value pairs
+delimited by newlines. For example, we would like to extract the
+`password` value from the following made-up input:
+
+```
+username = bob
+password = p@$$w0rd
+server = example.com
+```
+
+Unfortunately, the following pattern won't work the way we want
+because in generic mode, metavariables will only capture a single word
+(alphanumeric sequence):
+
+```
+password = $PASSWORD
+```
+
+This would match the input file but would assign the value `p`
+to `$PASSWORD` instead of the full value `p@$$w0rd`.
```

Comment:
Avoid would, could, may, should in documentation. It can express doubt and uncertainty. See: https://developers.google.com/style/word-list?hl=en#could

```suggestion
This pattern matches the input file but does not assign the value `p` to `$PASSWORD` instead of the full value `p@$$w0rd`.
```